node/src/Unison/ABT/Extra.hs

```diff
@@ -21,10 +21,12 @@ import qualified Data.Map as Map
 import qualified Data.Vector as Vector
 import qualified Unison.Digest as Digest
 
-hash :: forall f . (Foldable f, Digest.Digestable1 f) => Term f -> Digest.Hash
+-- | We ignore annotations in the `Term`, as these should never affect the
+-- meaning of the term.
```

Comment:
I was not positive about this decision, but the idea of leaving the annotation out of the hashes is that the annotation does not affect the meaning of a term, and the hash is meant to uniquely identify a meaning. Same reason that the free variables set doesn't affect hashing - it's just bookkeeping info.

Terms are already tied to a typechecker anyway, the typechecker is telling us what terms even make sense to run.

Anyway, I am still kind of skeptical that storing inferred types everywhere is a good idea. Besides the inefficiency, it also gets out of date in really unpredictable ways and seems error prone to use. You might modify a subterm, and now subterms very far from your original edit have incorrect annotations, depending on how typing info flows. Having to worry about whether I'm doing that does not sound fun...

If you only recreate the info on demand based on the current state of the world, there's no way it can ever be out of sync! :) As long as you can service these queries in an amount of time that a human perceives as being instantaneous, you are good.

Anyway, I don't want to discourage you from trying out building a fully-annotated tree if you're super interested in that. I'm just not personally sold on it... :) I see it being potentially useful as more of an efficiency thing - you ask the node to give you back an annotated tree, which lets you hover over multiple subexpressions to view their types, without a roundtrip to the node each time.